src/key.rs

```diff
@@ -0,0 +1,17 @@
+//! Defines the VerificationKey type used to verify Schnorr signatures.
```

Comment:
I think there's little need for `key` module if this can be defined in the `signature`. Because the VerificationKey type is only used in the signature protocol.

Reply:
got it! we do use the `VerificationKey` type for the vector `signtx_keys` in the `vm` module, but I think it would still make sense to define it in `signature` :+1:

Reply:
yeah, it's still logically only used to make/check the signatures, so it's a "signature subprotocol" entity, not a shared entity between two different subprotocols.